src/js/edu-benefits/components/MilitaryServiceFields.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+
+import ErrorableRadioButtons from '../../common/components/form-elements/ErrorableRadioButtons';
+import ErrorableTextInput from '../../common/components/form-elements/ErrorableTextInput';
+import GrowableTable from '../../common/components/form-elements/GrowableTable';
+
+import MilitaryServiceTour from './MilitaryServiceTour';
+import { createTour } from '../utils/veteran';
+
+import { validateIfDirty, isNotBlank, isValidYear, isValidSection } from '../utils/validations';
+import { yesNo } from '../utils/options-for-select';
+
+export default class MilitaryServiceFields extends React.Component {
+  render() {
+    const tourFields = [
+      'doNotApplyPeriodToSelected',
+      'applyToChapter30',
+      'applyToChapter1606',
+      'applyToChapter32',
+      'serviceBranch',
+      'toDate',
+      'fromDate',
+      'serviceStatus',
+      'involuntarilyCalledToDuty'
+    ];
+    const activeDutyQuestions = (
+      <div>
+        <ErrorableRadioButtons
+            errorMessage={validateIfDirty(this.props.data.currentlyActiveDuty.onTerminalLeave, isNotBlank) ? '' : 'Please select a response'}
+            label="Are you on terminal leave?"
+            name="onTerminalLeave"
+            options={yesNo}
+            value={this.props.data.currentlyActiveDuty.onTerminalLeave}
+            onValueChange={(update) => {this.props.onStateChange('currentlyActiveDuty.onTerminalLeave', update);}}/>
+        <ErrorableRadioButtons
+            errorMessage={validateIfDirty(this.props.data.currentlyActiveDuty.nonVaAssistance, isNotBlank) ? '' : 'Please select a response'}
+            label="Are you receiving, or do you anticipate receiving, any money (including but not limited to federal tuition assistance) from the armed forces or public health services for the course for which you have applied to the VA for education benefits?"
+            name="nonVaAssistance"
+            options={yesNo}
+            value={this.props.data.currentlyActiveDuty.nonVaAssistance}
+            onValueChange={(update) => {this.props.onStateChange('currentlyActiveDuty.nonVaAssistance', update);}}/>
+      </div>
+    );
+
+    return (<fieldset>
+      <legend>Military Service</legend>
+      <p>(<span className="form-required-span">*</span>) Indicates a required field</p>
+      <div className="input-section">
+        <p>If you graduated from a military service academy, what year did you graduate?</p>
+        <ErrorableTextInput
+            errorMessage={validateIfDirty(this.props.data, isValidYear) ? undefined : 'Please enter a valid year'}
```

Comment:
Ah, good point. The DateInput component puts `yyyy`, makes sense to do that here.